internal/broker/broker.go

```diff
@@ -204,24 +211,19 @@ func (b *Broker) GetAuthenticationModes(sessionID string, supportedUILayouts []m
 	_, err = os.Stat(session.cachePath)
 	tokenExists := err == nil
 
-	// Checks if the provider is accessible and if device authentication is supported
-	providerReachable := true
-	r, err := http.Get(strings.TrimSuffix(b.auth.providerURL, "/") + "/.well-known/openid-configuration")
-	// This means the provider is not available or something bad happened, so we assume no connection.
-	if err != nil || r.StatusCode != http.StatusOK {
-		providerReachable = false
+	endpoints := make(map[string]struct{})
+	if session.authCfg.provider != nil {
+		if session.authCfg.provider.Endpoint().DeviceAuthURL != "" {
```

Comment:
this is rather an `&&` in a single if, no?

Reply:
I was going for a && at first, but then wasn't sure whether Go evaluates the if lazily or not. Do you know?

Reply:
Hem, you could test in a playground :p

But yeah, to answer on that one, it is lazily (like most of the languages)